buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchNode.java

```diff
@@ -209,6 +210,7 @@ public class OpenSearchNode implements TestClusterConfiguration {
         setTestDistribution(TestDistribution.INTEG_TEST);
         setVersion(VersionProperties.getOpenSearch());
         this.zone = zone;
+        this.credentials.add(new HashMap<>());
```

Comment:
Why do we need to add an empty map here?

Reply:
Adding an empty map properly instantiates the object `this.credentials`. Because `this.credentials` is defined as `private final List<Map<String, String>> credentials = new ArrayList<>();` then we had a list of maps with no maps in it. This meant the straightforward logic here 

```
   if (System.getProperty("tests.opensearch.username") != null) {
            this.credentials.get(0).put("username", System.getProperty("tests.opensearch.username"));
            LOGGER.info("Overwriting username to: " + this.getCredentials().get(0).get("username"));
        }
        if (System.getProperty("tests.opensearch.password") != null) {
            this.credentials.get(0).put("password", System.getProperty("tests.opensearch.password"));
            LOGGER.info("Overwriting password to: " + this.getCredentials().get(0).get("password"));
        }
```
would require duplicate instantiation (if only one was set we could get NPE)  or a separate instantiation loop. I am not sure why this is a list of maps instead of just a map but this made the most sense to me.